### DISCUSSION TITLE
Unraid build?

### DISCUSSION THREAD

**Discussion by jlvpineiro** [immich-app/immich#116] [Q&A]
Can anyone help getting this to run in unraid? Been trying all day with nothing to show for it. Super frustrated because I really want to try it.

**Comment by alextran1502** [comment]
Hello,
I am not familiar with unraid. I have a plan in the pipeline to simplify the Docker image to upload to Unraid Community App.

In the meantime, can you describe what did you try to run in Unraid? Are you able to spin up a virtual machine, ideally running Ubuntu or Debian?

**Reply by jlvpineiro** [reply]
So I tried going into the docker tab and adding a container. I didn't know what information to put.

 ![Screenshot_20220405-172932_Edge.jpg](https://user-images.githubusercontent.com/48294459/161852957-f7397af6-7d20-4b1e-97ff-2a70774d8ee4.jpg)

I tried going into the command line, going into /mnt/user/appdata, cloning this repo, and then going to the docker folder, changing .env, and running it. If I docker-compose from the main immich folder, it says all my .env customizations are blank. If I do it from the docker folder, then it runs, but then once it's through, it gets stuck in a loop of saying it can't connect to the server, it can't authenticate my password, and it's retrying. If I try to connect, it gives me a 502 error. I don't have any logs, but I could reproduce the behavior fairly quickly.

I also tried running it in an Ubuntu vm, but it gave some permissions errors about python files when I tried docker-compose. I deleted the vm and didn't save any logs or anything.

**Reply by alextran1502** [reply]
I think the best way to try now is to use the Ubuntu VM. Let's try that route. Can you help me capture the error you are seeing in the VM?

**Reply by jlvpineiro** [reply]
Trying with the vm again, I got this error.
![image](https://user-images.githubusercontent.com/48294459/161868489-5a1f1fc5-0547-4857-af1d-d31d444c928d.png)

**Reply by alextran1502** [reply]
Can you check which `docker` and `docker-compose` versions are you running?

Can you try this method of installing `docker-compose`? https://docs.docker.com/compose/install/

**Reply by jlvpineiro** [reply]
Hi, I figured it out. Turned out I had to chmod 666 docker. Then it was giving me another error:

ERROR: for database  Cannot start service database: driver failed programming external connectivity on endpoint immich_postgres (91b4c76e0ac610238e8d6a11e6d2b2932294657196f4afa8da04cb6f4b3657e0): Error starting userland proxy: listen tcp4 0.0.0.0:5432: bind: address already in use

I had to kill the process using :5432, and then it finally ran. Then I had to make my vm a bridged connection and it finally worked. Now it works. Now I can finally try it out which is nice. I do need it running on Unraid without a vm in order to actually use it beyond just trying it out though, so I guess that's the next thing to figure out.

**Reply by alextran1502** [reply]
I am glad it is working now!

You can follow this [branch](https://github.com/alextran1502/immich/tree/deployment/containerize-app-into-single-dockerfile) to look at my work on a single docker image for Unraid

**Reply by hbooo** [reply]
Also looking for a single docker image for unraid but your link leads to a 404

**Reply by sohail579** [reply]
@alextran1502 just want to say thanks you have made something awesome here.  also looking forward for the unraid single docker image.  your link though is not valid.  I also have another question which i will raise in another post
